### ISSUE TITLE
Unknown global Ember.inject

### ISSUE THREAD

**Issue by Dhaulagiri** [ember-cli/ember-modules-codemod#34]
_Moved from https://github.com/ember-cli/ember-rfc176-data/issues/34_

If you have inject destructured like this the ember-modules-codemod fails stating this in an unkown global:

`const { inject } = Ember;`

This form of destructing does work:

`const { inject: { service } } = Ember;`

Can/should the first case be supported?

**Comment by rwjblue** [comment]
Yes, we should definitely handle this a bit better.

If the `Ember.computed` case that I mentioned in https://github.com/ember-cli/ember-rfc176-data/issues/34 is working properly, you should checkout https://github.com/ember-cli/ember-modules-codemod/blob/master/transform.js#L230-L270 to see what updates are required to make the same work for `Ember.inject`...

**Comment by Dhaulagiri** [comment]
Same issue also if you use `const { inject } = Ember;` with `inject.controller`.  Taking a look to see if I can't figure out how to fix...

**Comment by Dhaulagiri** [comment]
The `Ember.computed` case works ok because `import { computed } from '@ember/object';` is valid whereas `import { inject } from '@ember/service';` is not

**Comment by rwjblue** [comment]
Ah, that makes sense.  Though, I do think we should actually update it to import things directly.  I don't think we intend to allow the following (it only happens to work due to the way the globals are setup):

```js
import computed from '@ember/computed';

export default Component.extend({
  derp: computed.alias('otherThing.derp')
})
```

My suggestion would be to add some additional logic to handle these sorts of namespaces (`Ember.computed` and `Ember.inject` might be the only examples?) so that we transform:

```js
const { computed } = Ember;

export default Component.extend({
  derp: computed.alias('otherThing.derp')
})
```

Into:

```js
import { alias } from '@ember/computed';

export default Component.extend({
  derp: alias('otherThing.derp')
})
```

**Comment by Dhaulagiri** [comment]
👍 that seems prudent.  

Having never worked on a codemod before I'm honestly finding it a little hard to grok the best way to accomplish this, but I'll poke around at it some more.  It's not _super_ hard to manually migrate these either, but it'd obviously be nice if the codemod worked here as well.

**Comment by Dhaulagiri** [comment]
Although I think `import { inject } from '@ember/service';` should end up as `import { inject as service } from '@ember/service';` in the example above?

**Comment by rwjblue** [comment]
Ya, maybe name it `injectService`?

**Comment by Dhaulagiri** [comment]
Not sure if it matters, but the [docs](https://github.com/ember-cli/ember-rfc176-data) show them like this:

<img width="698" alt="screen shot 2017-08-12 at 15 31 56" src="https://user-images.githubusercontent.com/1672302/29244010-73bb8d62-7f73-11e7-9d8b-9df578366caf.png">

**Comment by Dhaulagiri** [comment]
You can ignore my last comment, I was doing it wrong.

```
import { inject } from '@ember/controller';
...
foo: inject('myControllerName') //works
foo: inject.controller('myControllerName') // does not work
```

**Comment by XaserAcheron** [comment]
Hmm, I've been doing some brainstorming on this, and it's a really weird case.

The tricky bit: if we have an instance of `import { inject } from Ember`, this means the codemod might need to replace it with either `import { inject } from '@ember/service'`, or `import { inject } from '@ember/controller'`, or _both_, and which of the two (or both) to use depends on the presence of any `inject.service` or `inject.controller` instances later on in the file.

Is there anything in the codemod currently that does this type of lookahead? This seems a bit can-of-worms-y, but having to do this case by-hand isn't handy (heh) either.

Also we'll need to figure out what to do in the case of injecting both a controller and a service in the same file, since the imports can't both be named `inject`. That seems like a change request for RFC176 itself, though.

**Comment by simonihmig** [comment]
This should fix it: #46
